File: Python_Prototype/algorithms/accumData/accumData.py
''' Originally developed in MATLAB by Fred Hickernell. Translated to python by Sou-Cheng T. Choi and Aleksei Sorokin '''
from abc import ABC, abstractmethod
from math import inf, nan
from numpy import array
from algorithms.util import univ_repr

class accumData(ABC):
    '''
    Accumulated data required in the computation of the integral
        Any sublcass of accumData must include:
            Methods: updateData(self, distribObj, fun_obj, decompType)
    '''
    def __init__(self):
        super().__init__()
        self.solution = nan  # solution
        self.stage = 'begin'  # stage of the computation, becomes 'done' when finished
        self.prevN = array([])  # new data will be based on (quasi-)random vectors indexed by
        self.nextN = array([])  # prevN + 1 to nextN
        self.timeUsed = array([])  # time used so far
        self.nSamplesUsed = array([])  # number of samples used so far
        self.confidInt = array([-inf, inf])  # error bound on the solution
        self.costF = array([])  # time required to compute function values
        self._timeStart = None # hidden/private

    # Abstract Methods
    @abstractmethod
    def updateData(self, distribObj, fun_obj, decompType):  # update the accumulated data
        pass
    
    # Magic Method
    def __repr__(self): return univ_repr(self,'accumData')
        

File: depricated/MATLAB_Prototype/accumData.m
classdef (Abstract) accumData
% Accumulated data required in the computation of the integral
properties
   solution = NaN %solution
   stage = 'begin' %stage of the computation, becomes 'done' when finished
   prevN %new data will be based on (quasi-)random vectors indexed by
   nextN %prevN + 1 to nextN
   timeUsed %time used so far
   nSamplesUsed %number of samples used so far
   confidInt = [-Inf Inf] %confidence interval for the solution
   costF %time required to compute function values
end

properties (Hidden)
   timeStart %starting time
end

methods (Abstract)
	updateData(obj, distribObj, fun_obj, decompType) %update the accumulated data
end
end
